packages/agreement-process/test/agreement.integration.test.ts

```diff
@@ -0,0 +1,738 @@
+        eserviceId,
+        descriptorId: descriptor.id,
+      };
+
+      await expect(
+        agreementService.createAgreement(apiAgreementPayload, authData)
+      ).rejects.toThrowError(
+        descriptorNotInExpectedState(eservice.id, descriptor.id, [
+          descriptorState.published,
+        ])
+      );
+    });
+
+    it("should throw an agreementAlreadyExists error when an Agreement in a conflicting state already exists for the same EService and consumer", async () => {
+      const consumer: Tenant = buildTenant();
+      const descriptor: Descriptor = buildDescriptorPublished();
+
+      const eservice = buildEService(generateId<EServiceId>(), consumer.id, [
+        descriptor,
+      ]);
+
+      const conflictingAgreement = {
+        ...buildAgreementWithValidCreationState(eservice.id, consumer.id),
+        state: randomArrayItem(agreementCreationConflictingStates),
+      };
+
+      await writeInReadmodel<Tenant>(consumer, tenants);
+      await writeInReadmodel<EService>(eservice, eservices);
+      await writeInReadmodel<Agreement>(conflictingAgreement, agreements);
+
+      const authData = getRandomAuthData(consumer.id);
+      const apiAgreementPayload: ApiAgreementPayload = {
+        eserviceId: eservice.id,
+        descriptorId: descriptor.id,
+      };
+
+      await expect(
+        agreementService.createAgreement(apiAgreementPayload, authData)
+      ).rejects.toThrowError(agreementAlreadyExists(consumer.id, eservice.id));
+    });
+
+    it("should throw a tenantIdNotFound error when the consumer Tenant does not exist", async () => {
+      const consumer: Tenant = buildTenant();
+      const descriptor: Descriptor = buildDescriptorPublished();
+
+      const eservice = buildEService(
+        generateId<EServiceId>(),
+        generateId<TenantId>(),
+        [descriptor]
+      );
+
+      await writeInReadmodel<EService>(eservice, eservices);
+
+      const authData = getRandomAuthData(consumer.id);
+      const apiAgreementPayload: ApiAgreementPayload = {
+        eserviceId: eservice.id,
+        descriptorId: descriptor.id,
+      };
+
+      await expect(() =>
+        agreementService.createAgreement(apiAgreementPayload, authData)
+      ).rejects.toThrowError(tenantIdNotFound(consumer.id));
+    });
+
+    it("should throw a missingCertifiedAttributesError error when the EService producer and Agreement consumer are different Tenants, and the consumer is missing a Descriptor certified Attribute", async () => {
+      const eserviceProducer: Tenant = buildTenant();
+
+      // Descriptor has two certified attributes
+      const certifiedDescriptorAttribute1: EServiceAttribute =
+        buildEServiceAttribute();
+      const certifiedDescriptorAttribute2: EServiceAttribute =
+        buildEServiceAttribute();
+
+      const descriptor = {
+        ...buildDescriptorPublished(generateId<DescriptorId>()),
+        attributes: {
+          certified: [
```

Comment:
Is there a reason why you do not pass the two attributes as params to `buildDescriptorPublished` as in other parts of the code?